common/src/main/java/com/microsoft/identity/common/internal/authorities/Authority.java

```diff
@@ -225,7 +226,17 @@ public static boolean isKnownAuthority(Authority authority) {
         }
 
         //Check if authority was added to configuration
-        knownToDeveloper = knownAuthorities.contains(authority);
+        if (authority.getKnownToDeveloper()) {
+            knownToDeveloper = true;
+        } else {
+            for (final Authority currentAuthority : knownAuthorities) {
+                if (currentAuthority.mAuthorityUrl != null && authority.getAuthorityURL() != null && authority.getAuthorityURL().getAuthority() != null &&
```

Comment:
nit : can we split the conditions each in different lines for readability

Reply:
Yes!